### DISCUSSION TITLE
Lama-cleaner and cleanup.pictures versus COTS for cleaning up wires

### DISCUSSION THREAD

**Discussion by mallman** [Sanster/IOPaint#273] [Q&A] [answered]
First of all, thank you so much for lama-cleaner. I'm impressed you've spent so much time porting all of those DL models! It must have been a lot of work.

I'm trying lama-cleaner on one of my landscape photos. I'm having particular trouble removing power cables with any of the DL models. In most cases, they don't do much at all—in fact they keep the wires just as-is. The best results I get in lama-cleaner is using the opencv `INPAINT_TELEA` model.

I get good results with three commercial apps I've tried: TouchRetouch from the AppStore, Affinity Photo's inpainting brush and Photoshop's content-aware fill. Unfortunately the last three are closed-source, and I have no idea how they work. I'm new to inpainting technology and don't know the different algorithms.

Any advice or ideas on how/why the DL models (like LaMa and MAT) erase lines? I'm attaching my test photo.

Thank you!

![_DSC0707](https://user-images.githubusercontent.com/833693/231205261-e535fabc-487f-4366-8f21-c240e49c2efb.jpeg)

**Comment by Sanster** [comment, marked as answer]
Thank you for the detailed description, I would like to know how you draw the mask? Below are the mask I drew and the result I get(using LaMa model).

|Mask I draw | Result |
|------------|--------|
|<img width="1214" alt="截屏2023-04-12 08 55 00" src="https://user-images.githubusercontent.com/3998421/231319834-0965ae2c-fc36-4be0-bf8a-197717f40d65.png">|![231205261-e535fabc-487f-4366-8f21-c240e49c2efb_cleanup](https://user-images.githubusercontent.com/3998421/231319658-b9c4fba4-488c-4faf-96f4-2201d161ab58.jpeg)|

**Reply by mallman** [reply]
Hi @Sanster. Thank you, this is very helpful. When I draw the mask like you did, I get the same results you do.

Some of the other tools I've used, like Affinity Photo, make it difficult (or impossible) to draw a complex mask (like three long lines) for inpainting, because it runs the inpainting algorithm immediately after every stroke. So when I do inpainting with AP, I draw short strokes incrementally. Affinity Photo's algorithm works well this way.

However, I see that LaMa works well when you draw the entire mask first, then run the inpainting operation. I used the "Manual Inpainting Mode" in Lama Cleaner to do this. I'm assuming you did the same thing—draw the entire mask first, then run the inpainter.

This gives me good insight into how to best use this tool.

Thanks again for your help! 👍
